documentation/projects/proposals/rekognition_data/20240423-implementation_plan_machine_generated_tags_api.md

```diff
@@ -0,0 +1,331 @@
+  creator-generated tags. This change would necessarily have an impact on search
+  performance, and since tags are one of the 3 text fields that get searched for
+  a query term, it would likely make a large portion of records no longer show
+  up in the results.
+- Many of our sources include
+  [GLAM institutions](<https://en.wikipedia.org/wiki/GLAM_(cultural_heritage)>).
+  These groups often go to great lengths to appropriately catalog and tag the
+  works that we ingest, and excluding those tags from being searched against
+  would be counter to that effort.
+- Machine-generated labeling is inherently biased, and may be incorrect in some
+  cases.
+
+#### Prefer the creator-generated tags, but boost on duplicate machine-generated tags
+
+This approach would likely need to leverage a custom
+[`rank_feature`](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-rank-feature-query.html)[^rank_feature]
+or
+[`function_score`](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-function-score-query.html).
+This would only give us the benefit of machine-generated tags _in cases where
+the machine tags matched the creator tags_. In other words, this would only
+improve the relevancy of results that were already present in search results for
+a given search term, it would not add any _new_ records to searches if the
+record did not already match for that term. In some of the exploratory analysis
+that was performed on the Rekognition-labeled images, maintainers discovered
+that the machine-generated tags were able to add relevant terms for images which
+did not have those tags already. For instance, an image of a cat with only the
+tag "best friend" from the creator would not show up under a search for "cat" in
+this case, even though the image has a cat in it and even if a machine-generated
+tag for "cat" were present.
+
+[^rank_feature]:
+    We already have a number of `rank_feature` fields, which could be used as a
+    basis for this approach:
+    <https://github.com/WordPress/openverse/blob/3f3376cfea8a4355229d4fb7d3a69a299de8cba4/ingestion_server/ingestion_server/es_mapping.py#L96-L104>
+
+#### Keep both tags and allow that to affect the document score
+
+This is the current approach, and also the simplest: leave the default
+`similarity` setting for `tags` and allow a result with duplicate tags to
+receive a higher score. This has the following advantages:
+
+- No action on the Elasticsearch index settings is needed (save for
+  documentation making this change explicit).
+- Machine-generated labels can help boost images with content that's relevant
+  for searches even if the other fields (`title`/`description`/`tags`) do not
+  mention the image content.
+- Records with duplicate machine- and creator-generated tags mean those tags are
+  likely quite relevant, and a higher score for them seems appropriate.
+- Records which don't have machine-generated labels are unaffected.
```

Comment:
This is not totally accurate. They won't be directly affected but they also won't receive a boost in searches, effectively making them less relevant.